update-repo.py

```diff
@@ -0,0 +1,140 @@
+import subprocess
+import sys
+import os
+
+import shutil
+
+
+"""
+This script streamlines the process of copying a .yml file into each workflows directory, 
+followed by creating a PR into the GitHub feedstock repository.
+
+How to use:
+
+python /path/.../update-repo.py <filepath-you-want-to-copy-in-your-local-machine> <filepath-in-the-github-repo>
+
+Workflow:
+
+- Walk through all directories to find corresponding filepath
+- Copy file to desired directories
+- Fetch the user's GitHub username using the GitHub CLI for authentication
+- Commit these changes, pushes them to GitHub, and creates a PR
+"""
+
+"""
+Utility Functions
+"""
+
+
+def run_command(command, cwd=None):
+    """Run a shell command in the specified directory."""
+    subprocess.run(
+        command,
+        check=True,
+        shell=True,
+        text=True,
+        cwd=cwd,
+    )
+
+
+"""
+Core Functionalities - find all desired directories, copy desired file, and create PR
+"""
+
+
+def copy_file(source_file, root_dir, dst_dir, username):
```

Comment:
since this func does more than copying, let's make the func name a bit more explicit? something along the lone of synching with main and create file. 

i think the word `create` is better than `copy` since when we create something, it is implied that it is copied since we aren't creating an empty file.

Reply:
what do you think?

Reply:
if you want to maintain "copy" another option is to separate the `run_command` section as a separate function such as `sync_with_main_branch`, etc. I think that would be better for for usage since we will use that syncing function often, for other potential tasks.